Source/rclUE/Public/Msgs/ROS2OdometryMsg.h

```diff
@@ -3,181 +3,182 @@
 
 #pragma once
 
-#include <CoreMinimal.h>
-
-#include "nav_msgs/msg/odometry.h"
-
 #include "Msgs/ROS2GenericMsg.h"
+#include "nav_msgs/msg/odometry.h"
 #include "rclcUtilities.h"
 
+#include <CoreMinimal.h>
+
 #include "ROS2OdometryMsg.generated.h"
 
 USTRUCT(Blueprintable)
 struct RCLUE_API FROSOdometry
 {
-	GENERATED_BODY()
+    GENERATED_BODY()
 
 public:
-	UPROPERTY(EditAnywhere, BlueprintReadWrite)
-	int header_stamp_sec;
-
-	unsigned int header_stamp_nanosec;
-
-	UPROPERTY(EditAnywhere, BlueprintReadWrite)
-	FString header_frame_id;
-
-	UPROPERTY(EditAnywhere, BlueprintReadWrite)
-	FString child_frame_id;
-
-	double pose_pose_position_x;
-
-	double pose_pose_position_y;
-
-	double pose_pose_position_z;
-
-	UPROPERTY(EditAnywhere, BlueprintReadWrite)
-	FQuat pose_pose_orientation;
-
-	TArray<double> pose_covariance;
-
-	UPROPERTY(EditAnywhere, BlueprintReadWrite)
-	FVector twist_twist_linear;
-
-	UPROPERTY(EditAnywhere, BlueprintReadWrite)
-	FVector twist_twist_angular;
-
-	TArray<double> twist_covariance;
-
-	
-
-	void SetFromROS2(const nav_msgs__msg__Odometry& in_ros_data)
-	{
-    	header_stamp_sec = in_ros_data.header.stamp.sec;
-
-		header_stamp_nanosec = in_ros_data.header.stamp.nanosec;
-
-		header_frame_id.AppendChars(in_ros_data.header.frame_id.data, in_ros_data.header.frame_id.size);
-
-		child_frame_id.AppendChars(in_ros_data.child_frame_id.data, in_ros_data.child_frame_id.size);
-
-		pose_pose_position_x = in_ros_data.pose.pose.position.x;
+    static constexpr int8 COVARIANCE_VALUES_NUM = 36;
+    UPROPERTY(EditAnywhere, BlueprintReadWrite)
+    int32 header_stamp_sec = 0;
 
-		pose_pose_position_y = in_ros_data.pose.pose.position.y;
+    UPROPERTY()
+    uint32 header_stamp_nanosec = 0;
```

Comment:
just curious : why is it better to add UPROPERTY() for these variables?

Reply:
Genuinely, this is just very much of my unconscious habits.. I vaguely remember some of my confusing experience with placing actors into the level then entering PIE that for primitive-typed members without `UPROPERTY`, their values would be all reset.
Since then I've been adding them for sake of confusion avoidance, but in this specific case, it could be not very essential.